util/changes.go

```diff
@@ -0,0 +1,35 @@
+package util
+
+import (
+	"reflect"
+)
+
+func RecursivelySetUndefinedFields(structToSet, structToDefaultFrom reflect.Value) {
```

Comment:
Needed to make some changes from the original implementation as this was panicking when used with the `UserSettings` Struct.